packages/babel-plugin-class-features/src/fields.js

```diff
@@ -0,0 +1,315 @@
+import { template, traverse, types as t } from "@babel/core";
+import { environmentVisitor } from "@babel/helper-replace-supers";
+import memberExpressionToFunctions from "@babel/helper-member-expression-to-functions";
+import optimiseCall from "@babel/helper-optimise-call-expression";
+
+export function buildPrivateNamesMap(props) {
+  const privateNamesMap = new Map();
+  for (const prop of props) {
+    if (prop.isPrivate()) {
+      const { name } = prop.node.key.id;
+      privateNamesMap.set(name, {
+        id: prop.scope.generateUidIdentifier(name),
+        static: !!prop.node.static,
+      });
+    }
+  }
+  return privateNamesMap;
+}
+
+export function buildPrivateNamesNodes(privateNamesMap, loose, state) {
+  const initNodes = [];
+
+  const looseHelper = () => state.addHelper("classPrivateFieldLooseKey");
+
+  for (const [name, { id, static: isStatic }] of privateNamesMap) {
+    if (loose) {
+      initNodes.push(
+        template.statement.ast`var ${id} = ${looseHelper()}("${name}")`,
```

Comment:
for my taste it would be better to inline `state.addHelper("classPrivateFieldLooseKey")` here, but do whatever u see fit